File: Roshaen/insertionSort.cpp
# include <iostream>
using namespace std;

int main(){
    cout<<"Enter the length of array"<<endl;
    int n;
    cin>>n;
    int arr[n];
    cout<<"Enter the elements of array"<<endl;
    for(int i =0; i<n; i++){
        cin>>arr[i];
    }

    for(int i=1; i<n;i++){
        int key = arr[i];
        int j = i-1;
        while (key<arr[j] && j>=0)
        {
            arr[j+1]=arr[j];
            j--;
        }
        arr[j+1]=key;
    }
    for (int i = 0; i < n; i++)
    {
        cout<<arr[i]<<" ";
    }
    
}

File: Vashi1/Insertion_sort.py
# Program to implement insertion sort in python
def insertionSort(arr):
  
    # Traverse through 1 to len(arr)
    for i in range(1, len(arr)):
  
        key = arr[i]
  
        # Move elements of arr[0..i-1], that are
        # greater than key, to one position ahead
        # of their current position
        j = i-1
        while j >=0 and key < arr[j] :
                arr[j+1] = arr[j]
                j -= 1
        arr[j+1] = key
  
  
arr = eval(input("Enter the list to be sorted: ")) 
insertionSort(arr)
print ("Sorted array is:")
for i in range(len(arr)):
    print ("%d" %arr[i])